climpred/utils.py

```diff
@@ -164,6 +181,28 @@ def convert_time_index(xobj, time_string, kind):
     return xobj
 
 
+def find_start_dates(init_pm, control, single_init):
+    """Find the same start dates for cftime single_init across different years in
+    control. Return control.time. Requires calendar=Datetime(No)Leap for consistent
+    `dayofyear`."""
```

Comment:
Can you clean this up a bit? I would also raise an error in the function if the calendar does not equal DatetimeNoLeap and say it's required for this function.

Reply:
how do I get the calendar type from an object?

Reply:
tried to add this exeception but couldnt do it

Reply:
found a related issue: https://github.com/pydata/xarray/issues/2416

Reply:
thanks to your help it works now. but isnt implemented nicely... you will see. I didnt have the nerve to implementing nicely

Reply:
Per our gitter chat, see `_retrieve_calendar` here: https://github.com/bradyrx/esmtools/blob/master/esmtools/temporal.py.  You can use `cftime_to_netcdf` in our constants for instance.

Reply:
I implemented it based on that suggestion. the problem is that single_init has no init dimension anymore, only init as coords. therefore the additional line